cli/admin/src/test/java/io/pravega/cli/admin/utils/TestUtils.java

```diff
@@ -290,4 +315,250 @@ public static void readAllEvents(String scope, String streamName, ClientFactoryI
+            } catch (Exception e) {
+                log.error("Exception occurred while starting bookKeeper service.", e);
+                this.close();
+                throw e;
+            }
+            this.bkService.set(this.bookKeeperServiceRunner);
+
+            // Create a ZKClient with a unique namespace.
+            this.baseNamespace = "pravega" + instanceId;
+            this.zkClient.set(CuratorFrameworkFactory
+                    .builder()
+                    .connectString("localhost:" + bkPort)
+                    .namespace(baseNamespace)
+                    .retryPolicy(new ExponentialBackoffRetry(1000, 10))
+                    .build());
+
+            this.zkClient.get().start();
+
+            logMetaNamespace = "segmentstore/containers" + instanceId;
+            this.bkConfig.set(BookKeeperConfig
+                    .builder()
+                    .with(BookKeeperConfig.ZK_ADDRESS, "localhost:" + bkPort)
+                    .with(BookKeeperConfig.ZK_METADATA_PATH, logMetaNamespace)
+                    .with(BookKeeperConfig.BK_LEDGER_PATH, ledgerPath)
+                    .build());
+        }
+
+        @Override
+        public void close() throws Exception {
+            val process = this.bkService.getAndSet(null);
+            if (process != null) {
+                process.close();
+            }
+
+            val bk = this.bookKeeperServiceRunner;
+            if (bk != null) {
+                bk.close();
+            }
+
+            val zkClient = this.zkClient.getAndSet(null);
+            if (zkClient != null) {
+                zkClient.close();
+            }
+        }
+    }
+
+    /**
+     * Creates a segment store.
+     */
+    public static class SegmentStoreRunner implements AutoCloseable {
+        private final int servicePort = io.pravega.test.common.TestUtils.getAvailableListenPort();
+        @Getter
+        private final int adminPort = io.pravega.test.common.TestUtils.getAvailableListenPort();
+        @Getter
+        private final ServiceBuilder serviceBuilder;
+        private final PravegaConnectionListener server;
+        private final StreamSegmentStore streamSegmentStore;
+        private final TableStore tableStore;
+
+        private AdminConnectionListener adminServer = null;
+
+        SegmentStoreRunner(StorageFactory storageFactory, BookKeeperLogFactory dataLogFactory, int containerCount, boolean enableAdminGateway)
+                throws DurableDataLogException {
+            ServiceBuilderConfig.Builder configBuilder = ServiceBuilderConfig
+                    .builder()
+                    .include(ServiceConfig.builder()
+                            .with(ServiceConfig.CONTAINER_COUNT, containerCount)
+                            .with(ServiceConfig.STORAGE_LAYOUT, StorageLayoutType.CHUNKED_STORAGE)
+                            .with(ServiceConfig.STORAGE_IMPLEMENTATION, ServiceConfig.StorageType.FILESYSTEM.name()))
```

Comment:
bit unsure this is always true. 
Is this class used by ECS and HDFS tests (directly or indirectly)?

Reply:
This class is only used in the admin CLI tests for the `list-chunks` test and the data recovery commands test.